torchtitan/config_manager.py

```diff
@@ -859,8 +1023,13 @@ def parse_args(self, args_list: list = sys.argv[1:]):
         self.args_dict = args_dict
 
         for k, v in args_dict.items():
-            class_type = type(k.title(), (), v)
-            setattr(self, k, class_type())
+            if k in self.__class__.__annotations__:  # Override dataclasses defaults
+                class_type = self.__class__.__annotations__[k]
+                setattr(self, k, class_type(**v))
+            else:
+                # Fallback to dynamic class creation for any new sections
```

Comment:
Or we could remove this entirely and enforce all config are typed in a dataclass.